fleetconfig-controller/internal/controller/addon.go

```diff
@@ -0,0 +1,218 @@
+	for i, ra := range requestedAddOns {
+		requestedVersionedNames[i] = fmt.Sprintf("%s-%s", ra.Name, ra.Version)
+	}
+
+	// Find addons that need to be created (present in requested, missing from created)
+	addonsToCreate := make([]*v1alpha1.AddOnConfig, 0)
+	for i, requestedName := range requestedVersionedNames {
+		if !slices.Contains(createdVersionedNames, requestedName) {
+			addonsToCreate = append(addonsToCreate, requestedAddOns[i])
+		}
+	}
+
+	// Find addons that need to be deleted (present in created, missing from requested)
+	addonsToDelete := make([]string, 0)
+	for _, createdName := range createdVersionedNames {
+		if !slices.Contains(requestedVersionedNames, createdName) {
+			addonsToDelete = append(addonsToDelete, createdName)
+		}
+	}
+
+	// do deletes first, then creates.
+	if len(addonsToDelete) > 0 {
+		err = handleAddonDelete(ctx, addonC, fc, addonsToDelete)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(addonsToCreate) > 0 {
+		err = handleAddonCreate(ctx, kClient, fc, addonsToCreate)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func handleAddonCreate(ctx context.Context, kClient client.Client, fc *v1alpha1.FleetConfig, addons []*v1alpha1.AddOnConfig) error {
+	// look up CM
+	addonConfigMap := corev1.ConfigMap{}
+	err := kClient.Get(ctx, types.NamespacedName{Name: addonConfigMapName, Namespace: fc.Namespace}, &addonConfigMap)
+	if err != nil {
+		return err
+	}
+
+	data := addonConfigMap.Data
+
+	// set up array of clusteradm addon create commands
+	for _, a := range addons {
+		// pull out manifests
+		manifests, ok := data[fmt.Sprintf("%s-%s", a.Name, a.Version)]
+		if !ok {
+			return fmt.Errorf("no manifests found for add-on %s version %s", a.Name, a.Version)
+		}
+
+		args := []string{
+			addon,
+			create,
+			a.Name,
+			fmt.Sprintf("--version=%s", a.Version),
+		}
+
+		switch checkManifestType(manifests) {
+		case manifestTypeHTTP:
+			// pass URL directly
+			args = append(args, fmt.Sprintf("--filename=%s", manifests))
+		case manifestTypeRaw:
+			filename, cleanup, err := file.TmpFile([]byte(manifests), "yaml")
+			if cleanup != nil {
+				defer cleanup()
+			}
+			if err != nil {
+				return err
+			}
+			args = append(args, fmt.Sprintf("--filename=%s", filename))
+		}
+
+		if a.HubRegistration {
+			args = append(args, "--hub-registration")
+		}
+		if a.Overwrite {
+			args = append(args, "--overwrite")
+		}
+		if a.ClusterRoleBinding != "" {
+			args = append(args, fmt.Sprintf("--cluster-role-bind=%s", a.ClusterRoleBinding))
+		}
+
+		cmd := exec.Command(clusteradm, args...)
+		out, err := exec_utils.CmdWithLogs(ctx, cmd, "waiting for 'clusteradm addon create' to complete...")
+		if err != nil {
+			return fmt.Errorf("failed to create addon: %v, output: %s", err, string(out))
+		}
+	}
+	return nil
+}
+
+func checkManifestType(manifests string) manifestType {
+	if strings.HasPrefix(manifests, "http") || strings.HasPrefix(manifests, "https") {
```

Comment:
nit: but can we be more specific in the prefix? ie would checking for `http://` and `https://` also be valid?